### PR TITLE
added new tests

### PR DESCRIPTION
**Pull request by aayeshaarshad**
Added unit tests for following templates 

- alertmanagerconfig
- backup
- certificate
- configmap
- forecastle
- grafanadashboard
- ingress
- networkpoliciy
- prometheusrule
- role
- rolebinding
- sercretproviderclass
- service
- servicemonitor
- vpa

### CODE REVIEW

**Comment by rasheedamir** [comment]
@aslafy-z can you please review this PR?

File: application/tests/alertmanagerconfig_test.yaml [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    capabilities:
+      apiVersions:
+        - monitoring.coreos.com/v1alpha1
+    asserts:
+      - equal:
+          path: metadata.labels.foo
+          value: bar
+      - equal:
+          path: metadata.labels.test
+          value: ing
```
Comment (aslafy-z) [inline application/tests/alertmanagerconfig_test.yaml:71]:
```suggestion
          value: ing

```

File: application/tests/backup_test.yaml [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+        enabled: true
+        excludedResources:
+          - secrets
+          - configmaps
+    asserts:
+      - equal:
+          path: spec.excludedResources
+          value:
+            - secrets
+            - configmaps
```
Comment (aslafy-z) [inline application/tests/backup_test.yaml:93]:
```suggestion
            - configmaps

```

File: application/tests/certificate_test.yaml [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+    asserts:
+      - equal:
+          path: spec.keystores.jks.create
+          value: true
+      - equal:
+          path: spec.keystores.jks.passwordSecretRef.key
+          value: my-jks-key
+      - equal:
+          path: spec.keystores.jks.passwordSecretRef.name
+          value: my-jks-name
```
Comment (aslafy-z) [inline application/tests/certificate_test.yaml:160]:
```suggestion
          value: my-jks-name

```

File: application/tests/grafanadashboard_test.yaml [hunk 4]
```diff
@@ -0,0 +1,210 @@
...
+    capabilities:
+      apiVersions:
+        - "grafana.integreatly.org/v1beta1"
+    asserts:
+      - equal:
+          path: spec.datasources[0].inputName
+          value: DS_PROMETHEUS
+      - equal:
+          path: spec.datasources[0].datasourceName
+          value: Prometheus
```
Comment (aslafy-z) [inline application/tests/grafanadashboard_test.yaml:210]:
```suggestion
          value: Prometheus

```

File: application/tests/ingress_test.yaml [hunk 5]
```diff
@@ -0,0 +1,192 @@
...
+        hosts:
+          - host: example.com
+            paths:
+              - path: /
+                serviceName: example-service
+                servicePort: http
+    asserts:
+      - equal:
+          path: spec.rules[0].http.paths[0].pathType
+          value: ImplementationSpecific
```
Comment (aslafy-z) [inline application/tests/ingress_test.yaml:192]:
```suggestion
          value: ImplementationSpecific

```

File: application/tests/networkpolicy_test.yaml [hunk 6]
```diff
@@ -0,0 +1,165 @@
...
+          custom: annotation
+        ingress: []
+        egress: []
+    asserts:
+      - equal:
+          path: metadata.labels.custom
+          value: label
+      - equal:
+          path: metadata.annotations.custom
+          value: annotation
```
Comment (aslafy-z) [inline application/tests/networkpolicy_test.yaml:165]:
```suggestion
          value: annotation

```

File: application/tests/prometheusrule_test.yaml [hunk 7]
```diff
@@ -0,0 +1,138 @@
...
+                  severity: warning
+                annotations:
+                  summary: "High request rate on {{ $labels.instance }}"
+    capabilities:
+      apiVersions:
+        - "monitoring.coreos.com/v1"
+    asserts:
+      - equal:
+          path: spec.groups[0].rules[0].expr
+          value: sum(rate(http_requests_total[5m])) by (instance) > 100
```
Comment (aslafy-z) [inline application/tests/prometheusrule_test.yaml:138]:
```suggestion
          value: sum(rate(http_requests_total[5m])) by (instance) > 100

```

File: application/tests/role_test.yaml [hunk 8]
```diff
@@ -0,0 +1,161 @@
...
+          - name: example
+            rules:
+              - apiGroups: ["*"]
+                resources: ["pods"]
+                verbs: ["get", "list"]
+      namespaceOverride: "custom-namespace"
+    asserts:
+      - equal:
+          path: metadata.namespace
+          value: custom-namespace
```
Comment (aslafy-z) [inline application/tests/role_test.yaml:161]:
```suggestion
          value: custom-namespace

```

File: application/tests/rolebinding_test.yaml [hunk 9]
```diff
@@ -0,0 +1,140 @@
...
+    set:
+      applicationName: my-app
+      rbac:
+        enabled: true
+        roles:
+          - name: role1
+          - name: role2
+    asserts:
+      - hasDocuments:
+          count: 2
```
Comment (aslafy-z) [inline application/tests/rolebinding_test.yaml:140]:
```suggestion
          count: 2

```

File: application/tests/servicemonitor_test.yaml [hunk 10]
```diff
@@ -0,0 +1,133 @@
...
+    set:
+      serviceMonitor:
+        enabled: true
+        endpoints: []
+      capabilities:
+        apiVersions:
+          - "monitoring.coreos.com/v1"
+    asserts:
+      - hasDocuments:
+          count: 0
```
Comment (aslafy-z) [inline application/tests/servicemonitor_test.yaml:133]:
```suggestion
          count: 0

```

**Review by aslafy-z — APPROVED** [review]
LGTM

**Pull request merged**